### PR TITLE
more robust edge filtering for the cases two subreads generating the same chimera connection

### PR DESCRIPTION
**Pull request by pb-jchin**
The updated case is more robust against such case as one of the
duplicated one will be removed. The repeat-bridge removal logic
will apply later if the such edges in the assembly graph is indeed
caused by repeats/chimera.

### CODE REVIEW

File: falcon_kit/mains/ovlp_to_graph.py [hunk 1]
```diff
@@ -1197,6 +1198,32 @@ def identify_spurs(ug, u_edge_data):
             break
     return ug2
 
+def remove_dup_simple_path(ug, u_edge_data):
+    # identify simple dup path
+    # if there are many multiple simple path of length connect s and t, e.g.  s->v1->t, and s->v2->t, we will only keep one
+    # Side-effect: Modifies u_edge_data
+    ug2 = ug.copy()
+    simple_edges = set()
+    dup_edges = {}
```
Comment (pb-cdunn) [inline falcon_kit/mains/ovlp_to_graph.py]: `defaultdict` can be your friend.

File: falcon_kit/mains/ovlp_to_graph.py [hunk 2]
```diff
@@ -1197,6 +1198,32 @@ def identify_spurs(ug, u_edge_data):
...
+            continue
+        if type_ == "simple":
+            if (s, t) in simple_edges:
+                dup_edges[ (s, t) ].append( v )
+            else:
+                simple_edges.add( (s, t) )
+                dup_edges[ (s, t) ] = [v]
+    for s, t in dup_edges:
+        vl = dup_edges[ (s, t) ]
+        vl.sort()
```
Comment (pb-cdunn) [inline falcon_kit/mains/ovlp_to_graph.py]: No interesting sort key?

File: falcon_kit/mains/ovlp_to_graph.py (hunk 1 above)
Comment (pb-jchin) [inline falcon_kit/mains/ovlp_to_graph.py]: understood.

**Comment by pb-jchin** [comment]
I will merge the code as other user needs a copy, I will merge it first, we can fix some other little things when it is necessary.

**Pull request merged**